test/parallel/test-assert-track-calls.js

```diff
@@ -117,6 +117,10 @@ process.on('exit', () => {
   trackedFunc();
   trackedFunc();
   trackedFunc();
+  const secondFunc = trackedFunc.add();
+  secondFunc();
+  const thirdFunc = trackedFunc.add({ expected: 1, mode: 'exact' });
+  thirdFunc();
```

Comment:
This part of the revised API just doesn't seem like an improvement to me and feels like a minor footgun. Specifically, I'd expect that it would be more likely for devs to end up creating multiple trackers rather than just one using this approach

Reply:
That seems a valid point to me. The updated API, which includes chaining through add() and dynamic mode switching, while providing more flexibility, may also make the API more complex and prone to misuse.

The worry here is that developers might find it easier to create multiple trackers instead of effectively using a single one due to the new approach's complexities.

In API design, achieving a balance between flexibility and simplicity is crucial, and it's essential to avoid creating features that might lead to unintentional errors.

When making changes to an API, it's important to carefully assess the trade-offs gather feedback from developers who will implement it in real-world situations, and consider revisiting the design to find a more straightforward and intuitive solution.